pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/impl/PersistentTopicsBase.java

```diff
@@ -2774,36 +2753,41 @@ protected void internalGetRetention(AsyncResponse asyncResponse, boolean applied
             throw new RestException(Status.PRECONDITION_FAILED,
                     "maxSubscriptionsPerTopic must be 0 or more");
         }
-        preValidation();
 
         TopicPolicies topicPolicies = getTopicPolicies(topicName).orElseGet(TopicPolicies::new);
         topicPolicies.setMaxSubscriptionsPerTopic(maxSubscriptionsPerTopic);
         return pulsar().getTopicPoliciesService().updateTopicPoliciesAsync(topicName, topicPolicies);
     }
 
     protected Optional<DispatchRate> internalGetReplicatorDispatchRate() {
-        preValidation();
         return getTopicPolicies(topicName).map(TopicPolicies::getReplicatorDispatchRate);
     }
 
     protected CompletableFuture<Void> internalSetReplicatorDispatchRate(DispatchRate dispatchRate) {
-        preValidation();
         TopicPolicies topicPolicies = getTopicPolicies(topicName).orElseGet(TopicPolicies::new);
         topicPolicies.setReplicatorDispatchRate(dispatchRate);
         return pulsar().getTopicPoliciesService().updateTopicPoliciesAsync(topicName, topicPolicies);
     }
 
-    private void preValidation() {
-        validateAdminAccessForTenant(namespaceName.getTenant());
-        validatePoliciesReadOnlyAccess();
+    protected void validateOwnership() {
         if (topicName.isGlobal()) {
             validateGlobalNamespaceOwnership(namespaceName);
         }
+        validateTopicOwnership(topicName, false);
+    }
+
+    protected void preValidateRead() {
+        validateOwnership();
         checkTopicLevelPolicyEnable();
     }
 
+    protected void preValidateWrite() {
+        validatePoliciesReadOnlyAccess();
```

Comment:
Seems we don't need to check this one? I noticed the policies read only check is to check if the broker allowed to do read-write operations based on the existence of a node in configuration metadata-store, but for the topic level policy, we are writing the policy data into a system topic.

Reply:
OK, so the read and write verification can be combined